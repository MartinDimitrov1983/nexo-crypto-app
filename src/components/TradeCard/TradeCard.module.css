.cardWrapper {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-bottom: 5px;
    border-radius: 5px;
    background-color: #fff;
    padding: 5px 10px;
    filter: drop-shadow(0 0 0.2rem rgb(193, 193, 193));
    width: 250px;
}

.cardContent {
    display: flex;
    width: 100%;
    justify-content: space-between;
}

.name {
    display: flex;
    justify-content: space-between;
    color: #5b6064;
    font-weight: 800;
    text-transform: capitalize;
}

.sell {
    color: red;
}

.buy {
    color: green;
}

.info {
    display: flex;
    margin-bottom: 0;
    padding-left: 0;
}
