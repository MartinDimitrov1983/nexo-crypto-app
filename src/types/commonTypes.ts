export interface Exchange {
    id: string;
    name: string;
    price: number;
    loading: Boolean;
}

export interface Sort {
    order: string;
    shouldSort: boolean;
}

export enum TRADING_PROVIDERS {
    binance = 'binance',
    bitfinex = 'bitfinex',
    huobi = 'huobi',
    kraken = 'kraken',
}

export enum TRADE_TYPES {
    buy = 'Buy',
    sell = 'Sell',
}

export enum SORTING_TYPES {
    ASCENDING = 'Ascending',
    DESCENDING = 'Descending',
}
