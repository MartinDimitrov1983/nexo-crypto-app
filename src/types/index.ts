export * from './commonTypes';
export * from './reduxTypes';
