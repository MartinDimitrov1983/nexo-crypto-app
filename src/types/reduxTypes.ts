export interface PlatformState {
    data: {
              symbol: string;
              price: string;
              error?: string;
          }
        | undefined;
    trades: [];
    isLoading: Boolean;
    isTradeLoading: Boolean;
}

export type TradeActionParams = {
    symbol: string;
    limit?: number;
};
