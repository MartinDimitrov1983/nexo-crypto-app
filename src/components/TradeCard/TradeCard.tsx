import React, { useState } from 'react';
import styles from './TradeCard.module.css';
import { AMOUNT, PRICE, TRADE_TYPE } from '../../constants';
import { TRADE_TYPES } from '../../types/commonTypes';

interface CardInfoProps {
    tradeType: TRADE_TYPES;
    price: string;
    amount: string;
}

const TradeCard = ({ tradeType, price, amount }: CardInfoProps) => {
    return (
        <div className={styles.cardWrapper}>
            <div className={styles.cardContent}>
                <div className={styles.name}>{TRADE_TYPE}</div>
                <div
                    className={`${
                        tradeType === TRADE_TYPES.sell
                            ? styles.sell
                            : styles.buy
                    }`}
                >
                    {tradeType}
                </div>
            </div>
            <div className={styles.cardContent}>
                <div className={styles.info}>{PRICE}</div>
                <div className={styles.info}>{price}</div>
            </div>
            <div className={styles.cardContent}>
                <div className={styles.info}>{AMOUNT}</div>
                <div className={styles.info}>{amount}</div>
            </div>
        </div>
    );
};

export default TradeCard;
