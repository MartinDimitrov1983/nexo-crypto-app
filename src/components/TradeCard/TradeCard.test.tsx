import React from 'react';
import { TRADE_TYPES } from '../../types/commonTypes';
import { render } from '@testing-library/react';

import TradeCard from './TradeCard';

test('Snapshot TradeCard', () => {
    const { asFragment } = render(
        <TradeCard tradeType={TRADE_TYPES.buy} price="test" amount="test" />,
    );
    expect(asFragment()).toMatchSnapshot();
});
