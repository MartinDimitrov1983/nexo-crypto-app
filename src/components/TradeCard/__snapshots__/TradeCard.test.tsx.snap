// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot TradeCard 1`] = `
<DocumentFragment>
  <div
    class="cardWrapper"
  >
    <div
      class="cardContent"
    >
      <div
        class="name"
      >
        Trade Type
      </div>
      <div
        class="buy"
      >
        Buy
      </div>
    </div>
    <div
      class="cardContent"
    >
      <div
        class="info"
      >
        Price
      </div>
      <div
        class="info"
      >
        test
      </div>
    </div>
    <div
      class="cardContent"
    >
      <div
        class="info"
      >
        Amount
      </div>
      <div
        class="info"
      >
        test
      </div>
    </div>
  </div>
</DocumentFragment>
`;
